src/git.rs

```diff
@@ -272,7 +275,12 @@ impl Repository {
         let head = self.head_oid()?;
 
         if head != original_head {
-            println!("Resetting index from {} to {}", original_head, head);
+            // This isn't strictly an error, but treating it as one for monitoring as it indicates
+            // that an error has occurred while processing the previous index operation.
+            eprintln!(
+                "at=error mod=git error=\"Resetting index from {} to {}\"",
+                original_head, head
+            );
```

Comment:
I'm wondering if we should use `error!(...)` instead, and use a custom `tracing` formatter to produce this output. `tracing` seems to explicitly be made for this kind of structured logging.